crates/esplora/src/blocking_ext.rs

```diff
@@ -339,59 +354,48 @@ fn fetch_txs_with_spks<I: IntoIterator<Item = ScriptBuf>>(
 /// Refer to [crate-level docs](crate) for more.
 fn fetch_txs_with_txids<I: IntoIterator<Item = Txid>>(
     client: &esplora_client::BlockingClient,
+    inserted_txs: &mut HashSet<Txid>,
     txids: I,
     parallel_requests: usize,
-) -> Result<TxGraph<ConfirmationBlockTime>, Error> {
-    enum EsploraResp {
-        TxStatus(TxStatus),
-        Tx(Option<Tx>),
-    }
-
-    let mut tx_graph = TxGraph::default();
-    let mut txids = txids.into_iter();
+) -> Result<tx_graph::Update<ConfirmationBlockTime>, Error> {
+    let mut update = tx_graph::Update::<ConfirmationBlockTime>::default();
+    // Only fetch for non-inserted txs.
+    let mut txids = txids
+        .into_iter()
+        .filter(|txid| !inserted_txs.contains(txid))
+        .collect::<Vec<Txid>>()
+        .into_iter();
     loop {
         let handles = txids
             .by_ref()
             .take(parallel_requests)
             .map(|txid| {
                 let client = client.clone();
-                let tx_already_exists = tx_graph.get_tx(txid).is_some();
                 std::thread::spawn(move || {
-                    if tx_already_exists {
-                        client
-                            .get_tx_status(&txid)
```

Comment:
It looks like we were calling `get_tx_status` if we already had it but this seems like a bug so you removed it?

Reply:
Yes. I wouldn't call it a bug. It's just that esplora always returns a tx with a status. If we already have a tx, that means we already have the status. No need to getting the status here again